server/src/main/java/org/opensearch/search/SearchModule.java

```diff
@@ -1297,6 +1299,10 @@ private SearchPlugin.ExecutorServiceProvider registerIndexSearcherExecutorProvid
         return provider;
     }
 
+    private void registerQueryCollectorContextSpec(List<SearchPlugin> plugins) {
```

Comment:
We can have a test in SearchModuleTests to test `registerQueryCollectorContextSpec`

Reply:
It is a private method.

Reply:
For private method, we can't add unit tests

Reply:
Yes :)
What I meant is adding a getter and a test in [testDoubleRegister](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/test/java/org/opensearch/search/SearchModuleTests.java#L121)
```
SearchPlugin registersDupeQueryCollectorContextSpec = new SearchPlugin() {
    @Override
    public List<Function<QueryCollectorArguments, QueryCollectorContextSpec>> getCollectorContextSpecFactories() {
        return List.of(args -> new QueryCollectorContextSpec("my_collector", c -> null));
    }
};
expectThrows(IllegalArgumentException.class, registryForPlugin(registersDupeQueryCollectorContextSpec));

```

Reply:
```
SearchPlugin registersDupeQueryCollectorContextSpec = new SearchPlugin() {
            @Override
            public List<QueryCollectorContextSpecFactory> getCollectorContextSpecFactories() {
                return List.of((searchContext, queryCollectorArguments) -> Optional.empty());
            }
        };
        expectThrows(IllegalArgumentException.class, registryForPlugin(registersDupeQueryCollectorContextSpec));
```

After doing some deep-dive, we don't need this test because the implemenatation lies in the plugin.

For example if
ns plugin --> provides a definition of this method

Then there can't be 2 instances of ns plugin running in the cluster. Therefore, there will no such case where we see this error.